### ISSUE TITLE
Generate schema from class 

### ISSUE THREAD

**Issue by 99ACA** [typestack/class-validator#770]
**I was trying to...**
Is there away to generate schema from class (that include all necessary decorators)

**The problem:**
- improve the runtime
- use the schema on json object

***Example:*** (pseudo code)
```javascript
export class Post {
  @Length(10, 20)
  title: string;

  @Contains('hello')
  text: string;
}
//create schema or validator
let postValidator = createValidator(Post)
//check is valid
postValidator.isValid({title:"hello"})
```

***Right now, I am using this function:***
```javascript
export async function validator<T extends {} = {}>(validation: new (...args: any[]) => T, input: Object, forbidUnknownValues: boolean = false, skipMissingProperties: boolean = false): Promise<boolean> {
    try {
        let validateObj = Object.assign(new validation(), input)
        await validateOrReject(validateObj, { validationError: { target: false }, forbidUnknownValues, skipMissingProperties });
        return true
    } catch (err) {
        console.warn(`Validation of ${validation.name} class failed. Errors: ${JSON.stringify(err)}`)
        return false
    }
}

validator(Post,{title:"hello"})
```

**Comment by ytetsuro** [comment]
Please see the following.

https://github.com/typestack/class-validator#defining-validation-schema-without-decorators

Your code an override validation property when using `Object.assing`.

**Comment by 99ACA** [comment]
@ytetsuro 

Thanks for replay,

I am trying to follow the principle single  source of truth (SSOT), meaning to have one class that include the validation rules and data presentation.
Regrading `Object.assing` the issue was with deep copy, so right now the code using `class-transformer`.

There are two mean cases:
-  create class and using it in the businesses flow
-  receive data over rest-api as json.

Currently the code using the following function that work fine.
```javascript 
import { validateOrReject } from "class-validator";
import { plainToClass } from 'class-transformer';

//Remark: to archive better performance - use schema
export async function validator<T extends {} = {}>(validation: new (...args: any[]) => T, input: Object, forbidUnknownValues: boolean = false, skipMissingProperties: boolean = false): Promise<{ isValid: boolean, error?: string }> {
    try {
        let validateObj = plainToClass(validation, input)
        await validateOrReject(validateObj, { validationError: { target: false }, forbidUnknownValues, skipMissingProperties });
        return { isValid: true }
    } catch (err) {
        let error = `Validation of ${validation.name} class failed. Validation errors: ${JSON.stringify(err)}`
        return { isValid: false, error }
    }
}
```

The issue that we trying to improve the runtime.
- Is there option to create json schema from class with all the relevant decorators? 
- Do you see any other issue with the function below?

**Comment by ytetsuro** [comment]
@99ACA 

Thanks, response.

What you want `json scheme` format?

If you want JavaScript plain object following see.

https://github.com/typestack/class-transformer#classtoplain

Maybe what you are looking for is not in class-validator, but in class-transform.

**Comment by 99ACA** [comment]
@ytetsuro

The logic should also validate the json, so the current solution:
1. Receive the json create class validator (using `class-transfomer`)
2. Verify the input (using `class-validator`) 

The current solution has performance overhead, the better solution could be:
1. Create json validator from the class decorators (only once, based on `class-validator` decorators)
2. Perform the validation using the json validator.

In the current solution will be able to remove the use of the `class-transfomer`

**Comment by vlapo** [comment]
I think schema-based validation will be deprecated in the future probably. We want to focus to decorators. But there are plenty of other libraries supporting json schema validation. For example https://github.com/sideway/joi.

Duplicate of https://github.com/typestack/class-validator/issues/129
